tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -638,6 +605,37 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext {
         val elemId = ArenaCell.idFromName(elemName)
```

Comment:
fail when `arraysEnabled` is true?

Reply:
Yes, but in the arrays encoding `TlaSetOper.in` should never be passed to `Z3SolverContext`.

Reply:
yes, that's why we should violate an assertion or throw an exception in this case

Reply:
With the arrays encoding, as soon as `getInPred` is called an exception is thrown, since `inCache` will be empty. Due to this I though there was no need to throw an exception here.